packages/analytics-core/src/plugins/destination.ts

```diff
@@ -77,9 +78,13 @@ export class Destination implements DestinationPlugin {
         this.schedule(this.config.flushIntervalMillis);
         return;
       }
+      this.queueBuffer.push(context);
       setTimeout(() => {
-        this.queue = this.queue.concat(context);
-        this.schedule(this.config.flushIntervalMillis);
+        const nextContext = this.queueBuffer.shift();
```

Comment:
Right, `event_id` might make more sense.

Reply:
`event_id` is optional. In this case, I use `Set` instead. Context reference is more accurate than `event_id`.

Reply:
I was hoping we'd find a way to simplify the design the destination plugin to only maintain 1 queue instead of 2, which we had decided to do during the design phase. I know using 2 queues are convenient but 1 queue would have been good design.

Reply:
I guess we can settle for this in the meantime and I'll put more thought to design this better, together with saving events (for browser).

Also I think we shouldn't be saving events using the in memory storage provider for server side since it doesn't really make sense or provide value. It will just take up more memory than it needs and capacity is cut in half.

Reply:
Sure. Let keep it for now. We can do optimization in a separate task.